src/fastalloc/tests.rs

```diff
@@ -102,10 +103,118 @@ fn test_debug_locations2() {
     );
     assert_eq!(result.debug_locations[1].0, 23);
     assert_eq!(result.debug_locations[1].1, ProgPoint::after(i(2)));
-    assert_eq!(result.debug_locations[1].2, ProgPoint::after(i(4)));
+    assert_eq!(result.debug_locations[1].2, ProgPoint::before(i(3)));
     assert!(matches!(result.debug_locations[1].3.as_stack(), Some(_)));
 }
 
+#[test]
+fn test_debug_locations_check1() {
+    let mach_env = mach_env(10);
+    let mut options = RegallocOptions::default();
+    options.validate_ssa = true;
+    options.algorithm = Algorithm::Fastalloc;
+    let mut f = RealFunction::new(vec![BlockBuildInfo {
+        insts: vec![
+            /* 0. */ vec![op(Def, 0, FixedReg(p(0)))],
+            /* 1. */
+            vec![
+                op(Def, 1, FixedReg(p(0))),
+                op(Use, 0, FixedReg(p(0))),
+                op(Use, 0, Reg),
+            ],
+            /* 2. */
+            vec![
+                op(Def, 2, FixedReg(p(8))),
+                op(Use, 0, FixedReg(p(2))),
+                op(Use, 1, FixedReg(p(0))),
+            ],
+            /* 3. */ vec![op(Def, 3, FixedReg(p(9))), op(Use, 0, FixedReg(p(9)))],
+        ],
+    }]);
+    f.debug_value_labels = vec![
+        (v(0), i(0), i(4), 32),
+        (v(2), i(2), i(4), 70),
+        (v(2), i(2), i(4), 71),
+        (v(3), i(3), i(4), 34),
+    ];
+    let mut result = run(&f, &mach_env, &options).unwrap();
+    /*
+    The correct debug_locations output
```

Comment:
Are we waiting for a fix before asserting this output instead? If so could we add a note here that this is to be asserted once we fix [thing]?